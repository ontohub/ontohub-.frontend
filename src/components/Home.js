import React from 'react'
import { withHeader } from '../helpers'
import Header from './Header'
import { Icon, List, Header as H, Message, Container } from 'semantic-ui-react'
import { css, sizes } from '../styles'

const Bold = (props) => (
  <span {...css({ fontWeight: 'bold' })}>
    {props.children}
  </span>
)

const Item = ({ icon, header, content }) => (
  <List.Item>
    <List.Icon name={icon} />
    <List.Content>
      <Bold>{header}</Bold>{' — '}{content}
    </List.Content>
  </List.Item>
)

const HomeHeader = () => (
  <Header>
    <Container {...css({ padding: sizes.headerPadding })}>
      <H as="h2" inverted>Welcome to Ontohub</H>
      <List size="large">
        <Item
          icon="fork"
          header="Open"
          content="based on open source software"
        />
        <Item
          icon="code"
          header="Flexible"
          content="supporting OWL, UML, FOL/TPTP, HOL/THF, and more"
        />
        <Item
          icon="sitemap"
          header="Distributed"
          content="OMS alignments, mappings, networks, combinations using DOL"
        />
      </List>
    </Container>
  </Header>
)

const Home = () => (
  <Container text>
    <Message icon size="tiny" warning>
      <Icon name="announcement" />
      <Message.Content>
        <Message.Header>Development preview</Message.Header>
        You are currently viewing an early development version
        of the new Ontohub website. It does not yet include much
        functionality. To use the stable version of Ontohub,
        please visit
        {' '}
        <a href="https://ontohub.org">ontohub.org</a>
        .
      </Message.Content>
    </Message>
  </Container>
)

const HomeWithHeader = withHeader(HomeHeader, Home)

export { HomeHeader, Home }

export default HomeWithHeader
