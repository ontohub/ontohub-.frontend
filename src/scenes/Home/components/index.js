export { Bold } from './Bold'
export { Item } from './Item'
export { Header } from './Header'
