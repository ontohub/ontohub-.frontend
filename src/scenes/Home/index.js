import React from "react";
import { Icon, Message, Container } from "semantic-ui-react";
import { Header } from "./components";

const Home = () => (
  <Container text>
    <Message icon size="tiny" warning>
      <Icon name="announcement" />
      <Message.Content>
        <Message.Header>Development preview</Message.Header>
        You are currently viewing an early development version of the new
        Ontohub website. It does not yet include much functionality. To use the
        stable version of Ontohub, please visit{" "}
        <a href="https://ontohub.org">ontohub.org</a>
        .
      </Message.Content>
    </Message>
  </Container>
);

export { Header, Home };

export default Home;
