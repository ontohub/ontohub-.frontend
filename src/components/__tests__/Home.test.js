import React from 'react'
import { Home, HomeHeader } from '../Home'
import renderer from 'react-test-renderer'

describe('Home', () => {
  const wrapper = renderer.create(<Home />)
  it('matches the snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})

describe('HomeHeader', () => {
  const wrapper = renderer.create(<HomeHeader />)
  it('matches the snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
