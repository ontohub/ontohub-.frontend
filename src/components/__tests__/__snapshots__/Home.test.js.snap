// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div
  className="ui text container"
>
  <div
    className="ui tiny icon warning message"
  >
    <i
      aria-hidden="true"
      className="announcement icon"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Development preview
      </div>
      You are currently viewing an early development version of the new Ontohub website. It does not yet include much functionality. To use the stable version of Ontohub, please visit
       
      <a
        href="https://ontohub.org"
      >
        ontohub.org
      </a>
      .
    </div>
  </div>
</div>
`;

exports[`HomeHeader matches the snapshot 1`] = `
<div
  data-css-e32ed5=""
>
  <div>
    <div
      className="ui container"
      data-css-g21272=""
    >
      <h2
        className="ui inverted header"
      >
        Welcome to Ontohub
      </h2>
      <div
        className="ui large list"
        role="list"
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <i
            aria-hidden="true"
            className="fork icon"
          />
          <div
            className="content"
          >
            <span
              data-css-l6c9n4=""
            >
              Open
            </span>
             — 
            based on open source software
          </div>
        </div>
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <i
            aria-hidden="true"
            className="code icon"
          />
          <div
            className="content"
          >
            <span
              data-css-l6c9n4=""
            >
              Flexible
            </span>
             — 
            supporting OWL, UML, FOL/TPTP, HOL/THF, and more
          </div>
        </div>
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <i
            aria-hidden="true"
            className="sitemap icon"
          />
          <div
            className="content"
          >
            <span
              data-css-l6c9n4=""
            >
              Distributed
            </span>
             — 
            OMS alignments, mappings, networks, combinations using DOL
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
